backend/src/controller/defenceController.ts

```diff
@@ -28,33 +28,74 @@ function handleDefenceActivation(req: DefenceActivateRequest, res: Response) {
 	const defenceId = req.body.defenceId;
 	const level = req.body.level;
 
-	if (defenceId && level !== undefined) {
-		// activate the defence
-		req.session.levelState[level].defences = activateDefence(
-			defenceId,
-			req.session.levelState[level].defences
+	if (!defenceId) {
+		sendErrorResponse(res, 400, 'Missing defenceId');
+		return;
+	}
+
+	if (level === undefined) {
+		sendErrorResponse(res, 400, 'Missing level');
+		return;
+	}
+
+	if (level < LEVEL_NAMES.LEVEL_1 || level > LEVEL_NAMES.SANDBOX) {
+		sendErrorResponse(res, 400, 'Invalid level');
+		return;
+	}
+
+	if (req.session.levelState[level].defences === undefined) {
+		sendErrorResponse(
+			res,
+			400,
+			'You cannot activate defences on this level, because it uses the default defences'
 		);
-		res.status(200).send();
-	} else {
-		res.status(400).send();
+		return;
 	}
+
+	if (req.session.levelState[3].defences === undefined) return; // this dumb type guard is needed, because we are accessing the array via enum, and so typescript doesn't account for the above type guards. This will be solved with #842. The below array indices won't need to be hardcoded either.
+
+	req.session.levelState[3].defences = activateDefence(
```

Comment:
I am confused. We have undefined defences on LEVEL_1 and LEVEL_2, but not on LEVEL_3 or SANDBOX, right? But here you are always activating a SANDBOX defence, never a LEVEL_3 defence.

Same for the other handlers, although you cannot configure or reset defences on LEVEL_3, only in SANDBOX.

Reply:
oh you're right. This is about to be fixed with the levels not being an enum any more